auth/auth-server/src/main.rs

```diff
@@ -363,33 +368,43 @@ fn with_tracing() -> warp::trace::Trace<impl Fn(warp::trace::Info) -> tracing::S
 }
 
 /// Handle a rejection from an endpoint handler
-async fn handle_rejection(err: Rejection) -> Result<impl Reply, Rejection> {
-    if let Some(api_error) = err.find::<ApiError>() {
-        let (code, message) = match api_error {
-            ApiError::InternalError(e) => {
-                error!("Internal server error: {e}");
-                (StatusCode::INTERNAL_SERVER_ERROR, DEFAULT_INTERNAL_SERVER_ERROR_MESSAGE)
-            },
-            ApiError::BadRequest(msg) => (StatusCode::BAD_REQUEST, msg.as_str()),
-            ApiError::TooManyRequests => (StatusCode::TOO_MANY_REQUESTS, "Rate limit exceeded"),
-            ApiError::Unauthorized => (StatusCode::UNAUTHORIZED, "Unauthorized"),
-        };
-
-        Ok(json_error(message, code))
+async fn handle_rejection(err: Rejection) -> Result<WithStatus<Json>, Rejection> {
```

Comment:
very cool to handle API layer errors by matching and a `json_error` constructor, it's giving me ideas of how to do this in js.